trait bounds not satisfied for OpNode DatabaseProvider

### Describe the bug

I'm trying to create a read-only DatabaseProvider for Unichain and am getting an error that trait bounds are not satisfied

### Steps to reproduce

In `cargo.toml` (reth 1.6.0):
```toml
[package]
name = "reth_issue"
version = "0.1.0"
edition = "2024"

[dependencies]
reth-optimism-chainspec = { path = "../../temp_repos/reth/crates/optimism/chainspec", features = [
    "superchain-configs",
] }
reth-optimism-node = { path = "../../temp_repos/reth/crates/optimism/node" }
eyre = "0.6"
reth-provider = { path = "../../temp_repos/reth/crates/storage/provider" }
```

In `main.rs`
```rust
use eyre::Result;
use reth_optimism_chainspec::generated_chain_value_parser;
use reth_optimism_node::OpNode;
use reth_provider::providers::ReadOnlyConfig;

fn main() -> Result<()> {
    let path = "/path/to/unichain/datadir";
    let chain_spec = generated_chain_value_parser("unichain")
        .ok_or_else(|| eyre::eyre!("Unichain not found in superchain registry"))?;

    let factory = OpNode::provider_factory_builder()
        .open_read_only(chain_spec, ReadOnlyConfig::from_datadir(path))?;

    let provider = factory.provider()?;

    Ok(())
}
```

`cargo check` I get error
```
error[E0599]: the method `provider` exists for struct `ProviderFactory<NodeTypesWithDBAdapter<OpNode, Arc<DatabaseEnv>>>`, but its trait bounds were not satisfied
  --> src/main.rs:14:28
   |
14 |     let provider = factory.provider()?;
   |                            ^^^^^^^^ method cannot be called due to unsatisfied trait bounds
   |
  ::: /Users/zach.wong/repos/temp_repos/reth/crates/node/types/src/lib.rs:52:1
   |
52 | pub struct NodeTypesWithDBAdapter<Types, DB> {
   | -------------------------------------------- doesn't satisfy `_: ProviderNodeTypes`
   |
  ::: /Users/zach.wong/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/op-alloy-consensus-0.18.14/src/transaction/envelope.rs:26:1
   |
26 | pub enum OpTxEnvelope {
   | --------------------- doesn't satisfy `_: Deref`, `_: ScaleValue`, `_: WrapperTypeDecode` or `_: WrapperTypeEncode`
   |
  ::: /Users/zach.wong/repos/temp_repos/reth/crates/optimism/primitives/src/receipt.rs:21:1
   |
21 | pub enum OpReceipt {
   | ------------------ doesn't satisfy `_: Deref`, `_: ScaleValue`, `_: WrapperTypeDecode` or `_: WrapperTypeEncode`
   |
   = note: the following trait bounds were not satisfied:
           `op_alloy_consensus::transaction::envelope::OpTxEnvelope: reth_db_api::scale::ScaleValue`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `reth_optimism_primitives::receipt::OpReceipt: reth_db_api::scale::ScaleValue`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `op_alloy_consensus::transaction::envelope::OpTxEnvelope: parity_scale_codec::codec::WrapperTypeEncode`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `op_alloy_consensus::transaction::envelope::OpTxEnvelope: parity_scale_codec::codec::WrapperTypeDecode`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `reth_optimism_primitives::receipt::OpReceipt: parity_scale_codec::codec::WrapperTypeEncode`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `reth_optimism_primitives::receipt::OpReceipt: parity_scale_codec::codec::WrapperTypeDecode`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `op_alloy_consensus::transaction::envelope::OpTxEnvelope: Deref`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`
           `reth_optimism_primitives::receipt::OpReceipt: Deref`
           which is required by `reth_node_types::NodeTypesWithDBAdapter<OpNode, Arc<reth_db::implementation::mdbx::DatabaseEnv>>: ProviderNodeTypes`

For more information about this error, try `rustc --explain E0599`.
```

### Node logs

```text

```

### Platform(s)

Mac (Apple Silicon)

### Container Type

Not running in a container

### What version/commit are you on?

1.6.0

### What database version are you on?

I pulled from main today (8/8)

### Which chain / network are you on?

Trying to use Unichain DB from a local copy of the db directory

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

n/a

### If you've built Reth from source, provide the full command you used

_No response_

### Code of Conduct

- [x] I agree to follow the Code of Conduct

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -162,6 +162,7 @@ members = [
     "examples/network-proxy/",
     "examples/node-custom-rpc/",
     "examples/node-event-hooks/",
+    "examples/op-db-access/",
     "examples/polygon-p2p/",
     "examples/rpc-db/",
     "examples/precompile-cache/",

--- a/examples/op-db-access/Cargo.toml
+++ b/examples/op-db-access/Cargo.toml
@@ -1,0 +1,11 @@
+[package]
+name = "example-op-db-access"
+version = "0.0.0"
+publish = false
+edition.workspace = true
+license.workspace = true
+
+[dependencies]
+reth-op = { workspace = true, features = ["node"] }
+
+eyre.workspace = true

--- a/examples/op-db-access/src/main.rs
+++ b/examples/op-db-access/src/main.rs
@@ -1,0 +1,23 @@
+//! Shows how manually access the database
+
+use reth_op::{chainspec::BASE_MAINNET, node::OpNode, provider::providers::ReadOnlyConfig};
+
+// Providers are zero cost abstractions on top of an opened MDBX Transaction
+// exposing a familiar API to query the chain's information without requiring knowledge
+// of the inner tables.
+//
+// These abstractions do not include any caching and the user is responsible for doing that.
+// Other parts of the code which include caching are parts of the `EthApi` abstraction.
+fn main() -> eyre::Result<()> {
+    // The path to data directory, e.g. "~/.local/reth/share/base"
+    let datadir = std::env::var("RETH_DATADIR")?;
+
+    // Instantiate a provider factory for Ethereum mainnet using the provided datadir path.
+    let factory = OpNode::provider_factory_builder()
+        .open_read_only(BASE_MAINNET.clone(), ReadOnlyConfig::from_datadir(datadir))?;
+
+    // obtain a provider access that has direct access to the database.
+    let _provider = factory.provider();
+
+    Ok(())
+}

